utilities/rpp-unittests/OCL/BatchPD_ocl_pkd3.cpp

```diff
@@ -0,0 +1,2130 @@
+       
+        char temp[1000];
+        strcpy(temp,src1);
+        strcat(temp, de->d_name);
+   
+        char temp_second[1000];
+        strcpy(temp_second,src1_second);
+        strcat(temp_second, de->d_name);    
+       
+        // if(ip_channel == 3)
+        // {
+            image = imread(temp, 1);
+            image_second = imread(temp_second, 1);
+        // }
+        // else
+        // {
+        //     image = imread(temp, 0);
+        //     image_second = imread(temp_second, 0);
+        // }
+       
+        Rpp8u *ip_image = image.data;
+        Rpp8u *ip_image_second = image_second.data;
+        for(j = 0 ; j < srcSize[i].height; j++)
+        {
+            for(int x = 0 ; x < srcSize[i].width ; x++)
+            {
+                for(int y = 0 ; y < ip_channel ; y ++)
+                {
+                    input[count + ((j * maxWidth * ip_channel) + (x * ip_channel) + y)] = ip_image[(j * srcSize[i].width * ip_channel) + (x * ip_channel) + y];
+                    input_second[count + ((j * maxWidth * ip_channel) + (x * ip_channel) + y)] = ip_image_second[(j * srcSize[i].width * ip_channel) + (x * ip_channel) + y];
+                }
+            }
+        }
+        i++;
+    }
+    closedir(dr2);
+
+    if (ip_bitDepth == 1)
+    {
+        Rpp8u *inputTemp, *input_secondTemp;
+        Rpp16f *inputf16Temp, *inputf16_secondTemp;
+       
+        inputTemp = input;
+        input_secondTemp = input_second;
+
+        inputf16Temp = inputf16;
+        inputf16_secondTemp = inputf16_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputf16Temp = (Rpp16f) *inputTemp / 255.0;
+            *inputf16_secondTemp = (Rpp16f) *input_secondTemp / 255.0;
+            inputTemp++;
+            inputf16Temp++;
+            input_secondTemp++;
+            inputf16_secondTemp++;
+        }
+    }
+    else if (ip_bitDepth == 2)
+    {
+        Rpp8u *inputTemp, *input_secondTemp;
+        Rpp32f *inputf32Temp, *inputf32_secondTemp;
+       
+        inputTemp = input;
+        input_secondTemp = input_second;
+
+        inputf32Temp = inputf32;
+        inputf32_secondTemp = inputf32_second;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputf32Temp = (Rpp32f) *inputTemp / 255.0;
+            *inputf32_secondTemp = (Rpp32f) *input_secondTemp / 255.0;
+            inputTemp++;
+            inputf32Temp++;
+            input_secondTemp++;
+            inputf32_secondTemp++;
+        }
+    }
+
+    cl_mem d_input,d_inputf16,d_inputf32, d_input_second, d_output,d_outputf16,d_outputf32;
+cl_platform_id platform_id;
+cl_device_id device_id;
+cl_context theContext;
+cl_command_queue theQueue;
+cl_int err;
+    err = clGetPlatformIDs(1, &platform_id, NULL);
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_GPU, 1, &device_id, NULL);
```

Comment:
Codacy found an issue: [Variable 'err' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)